external-crates/move/crates/move-ir-to-bytecode/src/compiler.rs

```diff
@@ -425,7 +427,14 @@ pub fn compile_module<'a>(
         struct_defs,
         function_defs,
     };
-    Ok((module, source_map))
+    set_module_version(&mut compiled_module, module.specified_version);
+    Ok((compiled_module, source_map))
+}
+
+fn set_module_version(module: &mut CompiledModule, version: Option<u32>) {
+    if let Some(version) = version.or_else(get_bytecode_version_from_env) {
```

Comment:
Hmmmm. Should this be `get_bytecode_version_from_env().or(version)`? 
Or maybe we just nuke `get_bytecode_version_from_env` since there is a compiler flag?

Reply:
Only reason I kept the env as well is because even though there's a compiler flag the flag is not surfaced up into the package system/the `move build` command. So if you ever wanted to build a package using the existing workflows and specify the version the only way of doing that is with the env variable (at least currently). 

> Should this be get_bytecode_version_from_env().or(version)

I went back and forth on this lol and which should override the other if both are set. I'm happy either way way so fine switching it around if that's preferable.

Reply:
I don't really care. Let's maybe just focus on propagating that flag and killing `get_bytecode_version_from_env`

Reply:
I think I'll keep `get_bytecode_version_from_env` at least in this PR since I think there's some nuance that's going to be needed in terms of how we expose the flag out into the build system (i.e., I don't think it should be exposed by default, but maybe only on debug/dev builds etc). Happy to lead the charge on that though (but in a separate PR :) )